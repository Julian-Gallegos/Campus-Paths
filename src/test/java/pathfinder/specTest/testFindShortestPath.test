# Find a path with Dijkstra's algorithm when
# there are four different paths

CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B

AddNode simpleGraph Aa
AddNode simpleGraph Ab
AddNode simpleGraph Ac

AddNode simpleGraph Ba
AddNode simpleGraph Bb
AddNode simpleGraph Bc

AddNode simpleGraph Ca
AddNode simpleGraph Cb
AddNode simpleGraph Cc

AddEdge simpleGraph A B 15.0

AddEdge simpleGraph A Aa 3.0
AddEdge simpleGraph Aa Ab 3.0
AddEdge simpleGraph Ab Ac 3.0
AddEdge simpleGraph Ac B 0.0

AddEdge simpleGraph A Ba 1.0
AddEdge simpleGraph Ba Bb 2.0
AddEdge simpleGraph Bb Bc 3.0
AddEdge simpleGraph Bc B 0.0

AddEdge simpleGraph A Ca 5.0
AddEdge simpleGraph Ca Cb 5.0
AddEdge simpleGraph Cb Cc 5.0
AddEdge simpleGraph Cc B 0.0

FindPath simpleGraph A B