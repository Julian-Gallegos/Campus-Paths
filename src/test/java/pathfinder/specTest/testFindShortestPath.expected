# Find a path with Dijkstra's algorithm when
# there are four different paths

created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph

added node Aa to simpleGraph
added node Ab to simpleGraph
added node Ac to simpleGraph

added node Ba to simpleGraph
added node Bb to simpleGraph
added node Bc to simpleGraph

added node Ca to simpleGraph
added node Cb to simpleGraph
added node Cc to simpleGraph

added edge 15.000 from A to B in simpleGraph

added edge 3.000 from A to Aa in simpleGraph
added edge 3.000 from Aa to Ab in simpleGraph
added edge 3.000 from Ab to Ac in simpleGraph
added edge 0.000 from Ac to B in simpleGraph

added edge 1.000 from A to Ba in simpleGraph
added edge 2.000 from Ba to Bb in simpleGraph
added edge 3.000 from Bb to Bc in simpleGraph
added edge 0.000 from Bc to B in simpleGraph

added edge 5.000 from A to Ca in simpleGraph
added edge 5.000 from Ca to Cb in simpleGraph
added edge 5.000 from Cb to Cc in simpleGraph
added edge 0.000 from Cc to B in simpleGraph

path from A to B:
A to Ba with weight 1.000
Ba to Bb with weight 2.000
Bb to Bc with weight 3.000
Bc to B with weight 0.000
total cost: 6.000